### PR TITLE
Us 52 login functionality

### PR DESCRIPTION
**Pull request by kwidzinskidawid**

### CODE REVIEW

File: src/main/java/auth/TdpRecruitmentPasswordStore.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
+package auth;
```
Comment (drauf) [inline src/main/java/auth/TdpRecruitmentPasswordStore.java:33]: we should keep the copyright notice, like [here](https://github.com/wojciechh-kainos/tdp-invest/blob/master/src/main/java/auth/TdpInvestPasswordStore.java)

File: src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+define(['angular', 'application/auth/tdprAuthModule', 'application/auth/services/tdprAuthService'], function (angular, tdprAuthModule) {
+    tdprAuthModule.controller("tdprLoginController", function ($scope, tdprAuthService, $state, Notification) {
+
+        tdprAuthService.clearCredentials();
+
+        $scope.login = function () {
+            tdprAuthService.login($scope.username, $scope.password)
+                .then(function (user) {
+                    console.log(user);
+                   // $window.location.href = "/";
```
Comment (danielr-kainos) [inline src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js]: remove `console.log()` and commented out code

File: src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+                   $state.go('tdpr.recruiter.home')
+                }, function (response) {
+                    Notification.error({message: "Wrong email or password.", delay: 3500});
```
Comment (drauf) [inline src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js]: shouldn't we remove delay, like in #116?

File: src/main/java/auth/TdpRecruitmentUnauthorizedHandler.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import io.dropwizard.auth.UnauthorizedHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+/**
+ * Created by dawidkw on 22/08/16.
+ */
```
Comment (drauf) [inline src/main/java/auth/TdpRecruitmentUnauthorizedHandler.java]: unnecessary comment

Comment (RogowskaMalgorzata) [reply src/main/java/auth/TdpRecruitmentUnauthorizedHandler.java]: We know, it all will be changed, including removing comments and changing how the view looks.

File: src/main/resources/assets/js/application/auth/services/tdprAuthService.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+                var user = res.data;
+
+                currentUser = {
+                    firstName: user.firstName,
+                    lastName: user.lastName,
+                    email: user.email,
+                    token: user.token,
+                    isRecruiter: user.admin
+                };
+                console.log(user);
```
Comment (drauf) [inline src/main/resources/assets/js/application/auth/services/tdprAuthService.js]: remove `console.log()`

File: src/main/resources/assets/js/application/tdprModuleConfig.js [hunk 6]
```diff
@@ -13,8 +24,14 @@ define(['angular'
                     }
                 }
         });
-
     });
 
+    function _isUserLoggedIn(tdprAuthService) {
+    console.log("cacaac");
```
Comment (drauf) [inline src/main/resources/assets/js/application/tdprModuleConfig.js]: :)

File: src/main/resources/assets/js/application/auth/services/tdprAuthService.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        }
+
+        service.isAuthenticated = function (role) {
+            var deferred = $q.defer();
+
+            if (service.isUserLoggedIn()) {
+                deferred.resolve();
+            } else {
+                $location.path('/login');
+                Notification.error({message: 'You need to sign in to view this page.', delay: 3500});
```
Comment (drauf) [inline src/main/resources/assets/js/application/auth/services/tdprAuthService.js]: shouldn't we remove delay, like in #116?

File: src/main/resources/assets/js/application/auth/services/tdprAuthService.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+        service.isUserLoggedIn = function() {
+            return currentUser.token ? true : false;
+        };
+
+        service.getCurrentUser = function() {
+            return currentUser;
+        }
+
+        service.isAuthenticated = function (role) {
```
Comment (drauf) [inline src/main/resources/assets/js/application/auth/services/tdprAuthService.js]: can we use the `currentUser.isRecruiter` property here, or do you want to distinguish between roles in a different feature branch?

File: src/main/java/auth/TdpRecruitmentPasswordStore.java (hunk 1 above)
Comment (drauf) [reply src/main/java/auth/TdpRecruitmentPasswordStore.java:33]: resolved :ok_hand:

File: src/main/resources/assets/js/application/auth/services/tdprAuthService.js (hunk 8 above)
Comment (kwidzinskidawid) [reply src/main/resources/assets/js/application/auth/services/tdprAuthService.js]: yes we will, this is prepared for future feature

File: src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js (hunk 2 above)
Comment (danielr-kainos) [inline src/main/resources/assets/js/application/auth/controllers/tdprLoginController.js]: remove console.log() and commented out code

File: src/main/java/auth/TdpRecruitmentAuthenticator.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+public class TdpRecruitmentAuthenticator implements Authenticator<BasicCredentials, Person> {
+
+	private final PersonDao personDao;
+
+	private final TdpRecruitmentPasswordStore passwordStore;
+
+	private static final Cache<String, Person> cache = CacheBuilder
+			.newBuilder()
+			.maximumSize(100)
+			.expireAfterAccess(10, TimeUnit.MINUTES)
```
Comment (rlewan) [inline src/main/java/auth/TdpRecruitmentAuthenticator.java:26]: These should be taken from `app_config.yml`.

File: src/main/java/dao/PersonDao.java [hunk 10]
```diff
@@ -35,4 +37,10 @@ public void deleteById(Long id) {
     }
 
     public void update(Person person){currentSession().update(person);}
+
+    public Person getUserByEmail(String email) {
```
Comment (rlewan) [inline src/main/java/dao/PersonDao.java]: For methods which return a single value and can possibly not find the result, it's a good practice to return an [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) value.

File: src/main/java/resources/PersonResource.java [hunk 11]
```diff
@@ -115,4 +115,15 @@ public Person updatePerson(Person person) {
         personDao.update(person);
         return person;
     }
+
+    @PUT
+    @Path("/{id}/switchAccountStatus")
+    @UnitOfWork
+    public Response switchAccountStatus(@PathParam("id") Long id) {
```
Comment (rlewan) [inline src/main/java/resources/PersonResource.java:151]: `activateAccount` would better suit method's intent.

File: src/main/java/resources/PersonResource.java [hunk 12]
```diff
@@ -115,4 +115,15 @@ public Person updatePerson(Person person) {
...
+
+        Person person = personDao.getById(id);
+        person.setActive(!person.getActive());
```
Comment (rlewan) [inline src/main/java/resources/PersonResource.java]: What if the person is not found?

File: src/main/java/resources/PersonResource.java (hunk 11 above)
Comment (r-kruk-kainos) [reply src/main/java/resources/PersonResource.java:151]: It's used for activating and deactivating accounts also, so I think that's ok.

Comment (rlewan) [reply src/main/java/resources/PersonResource.java:151]: You're right, my bad. Still, it's better to have 2 endpoints then, one which activates and one which deactivates.

File: src/main/java/auth/TdpRecruitmentAuthorizer.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+import io.dropwizard.auth.Authorizer;
+
+public class TdpRecruitmentAuthorizer implements Authorizer<Person> {
+	@Override
+	public boolean authorize(Person principal, String role) {
+
+		if(role.equals("recruiter")) {
+			return principal.getAdmin() == true ? true : false;
+		} else if(role.equals("interviewer")) {
+			return principal.getAdmin() == false ? true : false;
```
Comment (rlewan) [inline src/main/java/auth/TdpRecruitmentAuthorizer.java]: These ternary operators are redundant, it's enough to do `return principal.getAdmin()` and `return !principal.getAdmin()`.

Comment (kwidzinskidawid) [reply src/main/java/auth/TdpRecruitmentAuthorizer.java]: getAdmin seemed like it doesnt say thats a boolean value, should i change it anyway?

Comment (rlewan) [reply src/main/java/auth/TdpRecruitmentAuthorizer.java]: You can do `return principal.getAdmin() == true` and `return principal.getAdmin() == false`.

File: src/main/java/auth/TdpRecruitmentAuthenticator.java (hunk 9 above)
Comment (kwidzinskidawid) [reply src/main/java/auth/TdpRecruitmentAuthenticator.java:26]: it will be moved if there is enough time

**Pull request merged**